File: numass-core/src/main/java/inr/numass/data/api/SimpleNumassPoint.java
package inr.numass.data.api;

import hep.dataforge.meta.Meta;
import hep.dataforge.meta.MetaBuilder;
import hep.dataforge.utils.MetaHolder;

import java.util.ArrayList;
import java.util.Collection;
import java.util.Comparator;
import java.util.List;
import java.util.stream.Stream;

/**
 * A simple static implementation of NumassPoint
 * Created by darksnake on 08.07.2017.
 */
public class SimpleNumassPoint extends MetaHolder implements NumassPoint {
    private final List<NumassBlock> blocks;

    /**
     * Input blocks must be sorted
     * @param voltage
     * @param blocks
     */
    public SimpleNumassPoint(double voltage, Collection<? extends NumassBlock> blocks) {
        this.blocks = new ArrayList<>(blocks);
        this.blocks.sort(Comparator.comparing(NumassBlock::getStartTime));
        super.setMeta(new MetaBuilder("point").setValue(HV_KEY, voltage));
    }

    public SimpleNumassPoint(Meta meta, Collection<? extends NumassBlock> blocks) {
        super(meta);
        this.blocks = new ArrayList<>(blocks);
        this.blocks.sort(Comparator.comparing(NumassBlock::getStartTime));
    }

    @Override
    public Stream<NumassBlock> getBlocks() {
        return blocks.stream();
    }
}

File: numass-core/src/main/kotlin/inr/numass/data/api/SimpleNumassPoint.kt
package inr.numass.data.api

import hep.dataforge.meta.Meta
import hep.dataforge.meta.MetaBuilder
import hep.dataforge.meta.MetaHolder

/**
 * A simple static implementation of NumassPoint
 * Created by darksnake on 08.07.2017.
 */
class SimpleNumassPoint(override val blocks: List<NumassBlock>, meta: Meta) : MetaHolder(meta), NumassPoint {

    /**
     * Input blocks must be sorted
     * @param voltage
     * @param blocks
     */
    constructor(blocks: Collection<NumassBlock>, voltage: Double) :
            this(blocks.sortedBy { it.startTime }, MetaBuilder("point").setValue(NumassPoint.HV_KEY, voltage))

    init {
        if(blocks.isEmpty()){
            throw IllegalArgumentException("No blocks in collection")
        }
    }

}
